File: c/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include<math.h>

//initialize counter
int main(void)
{
float change;
do
{
    change=get_float("Need change for?: ");
}
while (change <= 0);

//change float to int
int money = round(change * 100);

int counter = 0;

while (money >= 25)
{
    money -= 25;
    counter ++;
};
while (money < 25 && money >= 10)
{
    money -= 10;
    counter ++;
};
while (money < 10 && money >= 5)
{
    money -= 5;
    counter ++;
};
while (money > 0)
{
    money -= 1;
    counter ++;
};

printf("%i\n", counter);
}

File: pset6/cash.py
from cs50 import get_float
from math import floor


def main():
    while True:
        dollars = get_float("Change owed: ")
        change = floor(dollars * 100)

        if change > 0:
            break

    quarters = change // 25
    dimes = (change % 25) // 10
    nickels = ((change % 25) % 10) // 5
    pennies = ((change % 25) % 10) % 5

    print(f"{quarters + dimes + nickels + pennies}")


if __name__ == "__main__":
    main()